scipy/sparse/csgraph/_tools.pyx

```diff
@@ -368,10 +368,10 @@ def csgraph_to_masked(csgraph):
 
     >>> csgraph_to_masked(graph)
     masked_array(
-      data=[[--, 1.0, 2.0, --],
-            [--, --, --, 1.0],
-            [--, --, --, 3.0],
-            [--, --, --, --]],
+      data=[[ --, 1.0, 2.0,  --],
+            [ --,  --,  --, 1.0],
+            [ --,  --,  --, 3.0],
+            [ --,  --,  --,  --]],
```

Comment:
Seems like the previous version was correct, cf failed CI https://dev.azure.com/scipy-org/SciPy/_build/results?buildId=14042&view=logs&jobId=ede89041-5e8e-5201-ce5c-f68c2cbf02dd&j=ede89041-5e8e-5201-ce5c-f68c2cbf02dd&t=bf6d2fae-c4b4-5aaf-e29f-d3092465da33

Reply:
I can rollback it, if needed

Reply:
As you want, we just need to make sure that the run `scipy.scipy (Main refguide_asv_check)` is passing when doing changes on examples.

Reply:
Actually, I prefer new version, so just keep it for now

Reply:
We can only accept the changes if the CI is happy with it.

Reply:
Seems a little strange (if I'm not mistaken), because azure job error says, that it expects new version, but gets previous
<img width="347" alt="image" src="https://user-images.githubusercontent.com/26326659/133936403-0384c309-1f71-47d8-b10d-194310748e35.png">

Reply:
One way of fixing this and other things like it is teaching `refguide_check` to recognize numpy masked arrays. Since the repr cannot be just eval'd, something along these lines of might work

```
In [34]: print(s)
masked_array(
      data=[[ --, 1.0, 2.0,  --],
            [ --,  --,  --, 1.0],
            [ --,  --,  --, 3.0],
            [ --,  --,  --,  --]],
      mask=[[ True, False, False,  True],
            [ True,  True,  True, False],
            [ True,  True,  True, False],
            [ True,  True,  True,  True]],
 fill_value=1e+20)

In [35]: y = eval(s.replace('--', 'np.nan'))    # <<<<<

In [36]: y
Out[36]: 
masked_array(
  data=[[--, 1.0, 2.0, --],
        [--, --, --, 1.0],
        [--, --, --, 3.0],
        [--, --, --, --]],
  mask=[[ True, False, False,  True],
        [ True,  True,  True, False],
        [ True,  True,  True, False],
        [ True,  True,  True,  True]],
  fill_value=1e+20)
```

Reply:
Could you guys check if  https://github.com/scipy/scipy/pull/14751 fixes the failure?

Reply:
I will try to merge your branch into mine and push it all

Reply:
@ev-br, after merging your changes Refguide Check passed without failure!
<img width="210" alt="image" src="https://user-images.githubusercontent.com/26326659/133952377-980e2cea-1517-4b75-b852-89689f36cdea.png">